lib/src/lints/avoid_unused_parameters/visitors/avoid_unused_parameters_visitor.dart

```diff
@@ -48,6 +55,11 @@ class AvoidUnusedParametersVisitor extends RecursiveAstVisitor<void> {
         parameters.parameters.isEmpty) {
       return;
     }
+
+    if (ignoredEntities.matchClass(node)) {
```

Comment:
I think the idea is to create a mechanism that will allow us to easilly reuse such code between different linter rules instead of just implementing it separately for each rule.

Reply:
It has to be used slightly differently in different rules. I.e. in this one, the check has to be performed in the visitor, instead of directly in the `run` function, and `matchClass` and `matchMethod` have to be used separately. 

Most of the logic (parsing and matching against the list) is still inside the `IgnoredEntitiesModel`. Having two methods like that seemed like the minimal amount of code that is still flexible enough to adapt to different rules.

---

I just checked, and I think this particular rule can be implemented without the visitor. I can try that, but it won't change the way the check is performed here.